static/app/views/eventsV2/utils.tsx

```diff
@@ -380,6 +381,34 @@ function generateAdditionalConditions(
   return conditions;
 }
 
+/**
+ * Discover queries can query either Errors, Transactions or a combination
+ * of the two datasets. This is a util to determine if the query will excusively
+ * hit the Transactions dataset.
+ */
+export function usesTransactionsDataset(eventView: EventView, yAxisValue: string[]) {
+  let usesTransactions: boolean = false;
+  const parsedQuery = new MutableSearch(eventView.query);
+  for (let index = 0; index < yAxisValue.length; index++) {
+    const yAxis = yAxisValue[index];
+    const aggregateArg = getAggregateArg(yAxis) ?? '';
+    if (isMeasurement(aggregateArg) || aggregateArg === 'transaction.duration') {
+      usesTransactions = true;
+      break;
+    }
+    const eventTypeFilter = parsedQuery.getFilterValues('event.type');
+    if (
+      eventTypeFilter.length > 0 &&
+      eventTypeFilter.every(filter => filter === 'transaction')
```

Comment:
should we also check if `!event.type:error`?

Reply:
I think since we're explicitly checking for `eventTypeFilter.length`, we don't need the other check!

Reply:
Ah no I see what you mean- assuming explicitly saying !event.type:error will mean transactions. I think we have other event types than errors and transactions (like default, csp etc), not sure if we collect metrics on those, do you happen to know?